azext_edge/tests/edge/support/create_bundle/test_mq_int.py

```diff
@@ -0,0 +1,118 @@
+from .helpers import check_name, check_non_custom_file_objs, get_file_map, run_bundle_command
+
+logger = get_logger(__name__)
+
+
+@pytest.mark.parametrize("mq_traces", [False, True])
+def test_create_bundle_mq(init_setup, tracked_files, mq_traces):
+    """Test for ensuring file names and content. ONLY CHECKS mq."""
+    mq_traces = True
+
+    ops_service = OpsServiceType.mq.value
+    command = f"az iot ops support create-bundle --mq-traces {mq_traces} --ops-service {ops_service}"
+    walk_result = run_bundle_command(command=command, tracked_files=tracked_files)
+    file_map = get_file_map(walk_result, ops_service, mq_traces=mq_traces)
+    traces = file_map.pop("trace", {})
+    assert "diagnostic_metrics" in file_map
+
+    check_mq_types(file_map)
+
+    expected_file_objs = {
+        "deployment": [
+            "aio-mq-diagnostics-service",
+            "aio-mq-operator"
+        ],
+        "pod": [
+            "aio-mq-diagnostics-probe",
+            "aio-mq-diagnostics-service",
+            "aio-mq-dmqtt-authentication",
+            "aio-mq-dmqtt-backend",  # could have multiple
+            "aio-mq-dmqtt-frontend",  # same
+            "aio-mq-dmqtt-health-manager",
+            "aio-mq-operator"
+        ],
+        "replicaset": [
+            "aio-mq-diagnostics-service",
+            "aio-mq-operator"
+        ],
+        "service": [
+            "aio-mq-diagnostics-service",
+            "aio-mq-dmqtt-authentication",
+            "aio-mq-dmqtt-backend",
+            "aio-mq-dmqtt-frontend",
+            "aio-mq-dmqtt-health-manager"
+        ],
+        "statefulset": [
+            "aio-mq-diagnostics-probe",
+            "aio-mq-dmqtt-authentication",
+            "aio-mq-dmqtt-backend",  # could have multiple
+            "aio-mq-dmqtt-frontend",
+            "aio-mq-dmqtt-health-manager"
+        ]
+    }
+
+    check_non_custom_file_objs(file_map, expected_file_objs)
+
+    # There is a chance that traces are not present even if mq_traces is true
+    if not mq_traces:
+        assert not traces
+
+    if traces:
+        id_check = {}
+        for file in traces:
+            assert file["action"] in ["connect", "ping", "puback", "publish", "subscribe", "unsubscribe"]
+            check_name(file["name"], expected_file_objs["pod"])
+
+            # should be a json for each pb
+            if file["identifier"] not in id_check:
+                id_check[file["identifier"]] = {}
+            assert file["extension"] not in id_check[file["identifier"]]
+            # ex: id_check["b9c3173d9c2b97b75edfb6cf7cb482f2"]["json"]
+            id_check[file["identifier"]][file["extension"]] = True
+
+        for extension_dict in id_check.values():
+            assert extension_dict.get("json")
+            assert extension_dict.get("pb")
+
+
+def check_mq_types(file_map):
+    # do we always have these? What cases do they have them vs not?
+    # how can names change?
+    for config in file_map["brokerauthentication"]:
```

Comment:
What should happen for effective testing is to query the cluster for expected resources, in this case mq CRs then run the support bundle and ensure relevant CRs per service are captured in the expected format.

On one hand there is vanilla `init` deployment state, on the other hand there is post-deployment AIO in use state.

Reply:
path 1: use kubenetes python sdk thing to get all resources
path 2: use kubectl to get resources + parse strings/json

going to try path 2 first